### ISSUE TITLE
Rethink writing of double quotes around value to in flatten_and_write

### ISSUE THREAD

**Issue by theskumar** [theskumar/python-dotenv#15]
See #13

**Comment by tino** [comment]
Yeah, they do make sense. The problem was that the tool loading the environment variables (not with python-dotenv) was tripping upon them. But I guess I better fix that tool :)

**Comment by hugochinchilla** [comment]
I use this library to mantain my dotenv files across a number of projects, but I have a problem with Docker and it's that for Docker an envfile gets parsed as if you split the lines by the equal sign, everything before the equal is the variable name, the rest is the value, the problem is that you end up with quotes in your values which breaks some applications.

I guess changing the behavior by default could be a breaking change, maybe a good compromise is to support a flag to set the behavior:

~~~
 --quote=all (current behavior)
 --quote=off (never add quotes)
 --quote="auto" (only add quotes for values containing spaces)
~~~

I would add the "auto" behavior to support sourcing dotenv files from bash, i use this a lot while debugging, the problem with unquoted strings containing whitespace is that bash thinks you are trying to run a command.

**Comment by theskumar** [comment]
Just released 0.4.0 with this functionality. :tada: 

@hugochinchilla are you able to install the new version and confirm it works for you. Thanks.

**Comment by hugochinchilla** [comment]
Works amazingly, thank you for merging so fast.

**Comment by theskumar** [comment]
Released 0.6.0 containing this change is on PYPI. Yay!

**Comment by hugochinchilla** [comment]
@theskumar I think you wanted to post your last comment on PR #30 ;)

Thanks for merging :)
